packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-inline-checkout-button-strategy.ts

```diff
@@ -153,21 +151,36 @@ export default class PaypalCommerceInlineCheckoutButtonStrategy implements Check
         const orderDetails = await actions.order.get();
 
         if (cart.lineItems.physicalItems.length > 0) {
-            const address = this._getValidAddress(
-                orderDetails.payer.name,
-                orderDetails.payer.email_address,
-                orderDetails.purchase_units[0].shipping.address
-            );
+            const { payer, purchase_units } = orderDetails;
+            const shippingAddress = purchase_units?.[0]?.shipping?.address || {};
+
+            const address = this._getAddress({
+                firstName: payer.name.given_name,
+                lastName: payer.name.surname,
+                email: payer.email_address,
+                address1: shippingAddress.address_line_1,
+                city: shippingAddress.admin_area_2,
+                countryCode: shippingAddress.country_code,
+                postalCode: shippingAddress.postal_code,
+                stateOrProvinceCode: shippingAddress.admin_area_1,
+            });
 
             await this._store.dispatch(this._billingAddressActionCreator.updateAddress(address));
             await this._store.dispatch(this._consignmentActionCreator.updateAddress(address));
             await this._updateOrder();
         } else {
-            const address = this._getValidAddress(
-                orderDetails.payer.name,
-                orderDetails.payer.email_address,
-                orderDetails.payer.address
-            );
+            const { payer } = orderDetails;
+
+            const address = this._getAddress({
```

Comment:
Do we need to care about address_line_2?

Reply:
Nope, because PayPal didn't pass the data to fill `address_line_2` on our side